<a href="https://colab.research.google.com/github/LegendSeyi/Thai_Road_Accident_Data_Analysis/blob/main/Thai_Road_Accident_data_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **THAI ROAD ACCIDENT ANALYSIS**
#### **_By Adeyemi Oluwaseyi Emmanuel_**

[Click here to check Linkedin profile](https://www.linkedin.com/in/oluwaseyi-adeyemi-33b1ab197/)

[Twitter link](https://twitter.com/AmLegendseyi)

## **Introduction**

This dataset provides comprehensive statistics on recorded road accidents in Thailand, spanning from approximately 2019 to 2022. The data was sourced from raw information provided by the Office of the Permanent Secretary, Ministry of Transport, which is also utilized in this public dashboard for easier access and visualization. The dataset encompasses various aspects of road accidents and aims to shed light on the trends and patterns within this critical area of concern, analysis of this data could be crucial in guiding road safety policies and measures👍.

We import our required libraries

In [131]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [132]:
## creating a dictionary in python and converting it to a dataframe
tb = {'Column':['acc_code',
    'incident_datetime',
    'incident_datetime',
    'report_datetime',
    'province_th',
    'province_en',
    'agency',
    'route',
    'vehicle_type',
    'presumed_cause',
    'accident_type',
    'number_of_vehicles_involved',
    'number_of_fatalities',
    'number_of_injuries',
    'weather_condition',
    'latitude',
    'longitude',
    'road_description',
    'slope_description'],
    'Description':['The accident code or identifier',
    'The date and time of the accident occurrence',
    'The date and time when the accident was reported',
    'The name of the province in Thailand, written in Thai',
    'The name of the province in Thailand, written in English',
    'The government agency responsible for the road and traffic management',
    'The route or road segment where the accident occurred',
    'The type of vehicle involved in the accident',
    'The presumed cause or reason for the accident',
    'The type or nature of the accident',
    'The number of vehicles involved in the accident',
    'The number of fatalities resulting from the accident',
    'The number of injuries resulting from the accident',
    'The weather condition at the time of the accident',
    'The weather condition at the time of the accident',
    'The latitude coordinate of the accident location',
    'The longitude coordinate of the accident location',
    'The description of the road type or configuration where the accident occurred',
    'The description of the slope condition at the accident location']
     }

In [133]:
## Converting the dictionary to a data frame
dico = pd.DataFrame(tb)

In [134]:
dico

Column  \
0                      acc_code   
1             incident_datetime   
2             incident_datetime   
3               report_datetime   
4                   province_th   
5                   province_en   
6                        agency   
7                         route   
8                  vehicle_type   
9                presumed_cause   
10                accident_type   
11  number_of_vehicles_involved   
12         number_of_fatalities   
13           number_of_injuries   
14            weather_condition   
15                     latitude   
16                    longitude   
17             road_description   
18            slope_description   

                                          Description  
0                     The accident code or identifier  
1        The date and time of the accident occurrence  
2    The date and time when the accident was reported  
3   The name of the province in Thailand, written ...  
4   The name of the province in Thailand, written ...  
5   The government agency responsible for the road...  
6   The route or road segment where the accident o...  
7        The type of vehicle involved in the accident  
8       The presumed cause or reason for the accident  
9                  The type or nature of the accident  
10    The number of vehicles involved in the accident  
11  The number of fatalities resulting from the ac...  
12  The number of injuries resulting from the acci...  
13  The weather condition at the time of the accident  
14  The weather condition at the time of the accident  
15   The latitude coordinate of the accident location  
16  The longitude coordinate of the accident location  
17  The description of the road type or configurat...  
18  The description of the slope condition at the ...

The above table is the describtion of each columns in the dataset....

With this information, we can now import our daraset for the analysis

## **IMPORTING THE DATA SET**

In [135]:
# We will use the pandas library to import our dataset

In [136]:
# We can import our dataset from github


url = 'https://raw.githubusercontent.com/LegendSeyi/Dataset/main/thai_road_accident_2019_2022.csv'

pd.set_option('display.max_columns', None)
data = pd.read_csv(url, encoding= 'unicode_escape')


In [137]:
data.head()

acc_code    incident_datetime      report_datetime  \
0       571905  2019-01-01 00:00:00  2019-01-02 06:11:00   
1      3790870  2019-01-01 00:03:00  2020-02-20 13:48:00   
2       599075  2019-01-01 00:05:00  2019-01-01 10:35:00   
3       571924  2019-01-01 00:20:00  2019-01-02 05:12:00   
4       599523  2019-01-01 00:25:00  2019-01-04 09:42:00   

                                     province_th          province_en  \
0                             ลพบุรี               Loburi   
1              อุบลราชธานี     Ubon Ratchathani   
2  ประจวบคีรีขันธ์  Prachuap Khiri Khan   
3                    เชียงใหม่           Chiang Mai   
4                    นครสวรรค์         Nakhon Sawan   

                      agency  \
0  department of rural roads   
1     department of highways   
2     department of highways   
3  department of rural roads   
4     department of highways   

                                               route           vehicle_type  \
0  แยกทางหลวงหมายเà...             motorcycle   
1  เดชอุดม - อุบลราชà...  private/passenger car   
2  ปราณบุรี - ปากน้ำà...             motorcycle   
3  เชื่อมทางหลวงหมà...             motorcycle   
4         เกยไชย - ศรีมงคล  private/passenger car   

                                  presumed_cause  \
0         driving under the influence of alcohol   
1                                       speeding   
2                                       speeding   
3         driving under the influence of alcohol   
4  cutting in closely by people/vehicles/animals   

                        accident_type  number_of_vehicles_involved  \
0                               other                            1   
1    rollover/fallen on straight road                            1   
2  head-on collision (not overtaking)                            2   
3                               other                            1   
4    rollover/fallen on straight road                            1   

   number_of_fatalities  number_of_injuries weather_condition   latitude  \
0                     0                   2             clear  14.959105   
1                     0                   2             clear  15.210738   
2                     1                   0             clear  12.374259   
3                     0                   1             clear  18.601721   
4                     0                   0             clear  15.866389   

    longitude road_description slope_description  
0  100.873463    straight road          no slope  
1  104.862689    straight road          no slope  
2   99.907949       wide curve        slope area  
3   98.804204    straight road          no slope  
4  100.590010    straight road          no slope

We have imported our data, so we can check the size of the dataset

In [138]:
data.shape

(81735, 18)

We have 81735 row and 18 columns in the dataset

In [139]:
data.columns

Index(['acc_code', 'incident_datetime', 'report_datetime', 'province_th',
       'province_en', 'agency', 'route', 'vehicle_type', 'presumed_cause',
       'accident_type', 'number_of_vehicles_involved', 'number_of_fatalities',
       'number_of_injuries', 'weather_condition', 'latitude', 'longitude',
       'road_description', 'slope_description'],
      dtype='object')

we generated the list of columns available, but out first column is not properly named..

In [140]:
# we rename the first coloumn from ''acc_code' to 'acc_code'
data.columns = [col.replace("", "") for col in data.columns]

In [141]:
data.head()

acc_code    incident_datetime      report_datetime  \
0    571905  2019-01-01 00:00:00  2019-01-02 06:11:00   
1   3790870  2019-01-01 00:03:00  2020-02-20 13:48:00   
2    599075  2019-01-01 00:05:00  2019-01-01 10:35:00   
3    571924  2019-01-01 00:20:00  2019-01-02 05:12:00   
4    599523  2019-01-01 00:25:00  2019-01-04 09:42:00   

                                     province_th          province_en  \
0                             ลพบุรี               Loburi   
1              อุบลราชธานี     Ubon Ratchathani   
2  ประจวบคีรีขันธ์  Prachuap Khiri Khan   
3                    เชียงใหม่           Chiang Mai   
4                    นครสวรรค์         Nakhon Sawan   

                      agency  \
0  department of rural roads   
1     department of highways   
2     department of highways   
3  department of rural roads   
4     department of highways   

                                               route           vehicle_type  \
0  แยกทางหลวงหมายเà...             motorcycle   
1  เดชอุดม - อุบลราชà...  private/passenger car   
2  ปราณบุรี - ปากน้ำà...             motorcycle   
3  เชื่อมทางหลวงหมà...             motorcycle   
4         เกยไชย - ศรีมงคล  private/passenger car   

                                  presumed_cause  \
0         driving under the influence of alcohol   
1                                       speeding   
2                                       speeding   
3         driving under the influence of alcohol   
4  cutting in closely by people/vehicles/animals   

                        accident_type  number_of_vehicles_involved  \
0                               other                            1   
1    rollover/fallen on straight road                            1   
2  head-on collision (not overtaking)                            2   
3                               other                            1   
4    rollover/fallen on straight road                            1   

   number_of_fatalities  number_of_injuries weather_condition   latitude  \
0                     0                   2             clear  14.959105   
1                     0                   2             clear  15.210738   
2                     1                   0             clear  12.374259   
3                     0                   1             clear  18.601721   
4                     0                   0             clear  15.866389   

    longitude road_description slope_description  
0  100.873463    straight road          no slope  
1  104.862689    straight road          no slope  
2   99.907949       wide curve        slope area  
3   98.804204    straight road          no slope  
4  100.590010    straight road          no slope

So we have successfully renamed the column

SO we want to generate the info of the dataset

In [142]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 81735 entries, 0 to 81734
Data columns (total 18 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   acc_code                     81735 non-null  int64  
 1   incident_datetime            81735 non-null  object 
 2   report_datetime              81735 non-null  object 
 3   province_th                  81735 non-null  object 
 4   province_en                  81735 non-null  object 
 5   agency                       81735 non-null  object 
 6   route                        81735 non-null  object 
 7   vehicle_type                 81735 non-null  object 
 8   presumed_cause               81735 non-null  object 
 9   accident_type                81735 non-null  object 
 10  number_of_vehicles_involved  81735 non-null  int64  
 11  number_of_fatalities         81735 non-null  int64  
 12  number_of_injuries           81735 non-null  int64  
 13  weather_conditio

We can observe that we have different data types such ad float, int, object

In [143]:
# code to generate the number of null values present in the columns
data.isnull().sum()

acc_code                         0
incident_datetime                0
report_datetime                  0
province_th                      0
province_en                      0
agency                           0
route                            0
vehicle_type                     0
presumed_cause                   0
accident_type                    0
number_of_vehicles_involved      0
number_of_fatalities             0
number_of_injuries               0
weather_condition                0
latitude                       359
longitude                      359
road_description                 0
slope_description                0
dtype: int64

we can observe that column latitude and longitude contain the same number of missing values

## **ANALYSING EACH COLUMN**

### **Incident_datetime column**

In [144]:
# We start by analysing the 'incident datetime' column , we convert the type of column into a datetime
#data['incident_datetime'] = pd.to_datetime(data['incident_datetime'])

In [145]:
data['incident_datetime'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 81735 entries, 0 to 81734
Series name: incident_datetime
Non-Null Count  Dtype 
--------------  ----- 
81735 non-null  object
dtypes: object(1)
memory usage: 638.7+ KB


The column has been converted to a datetime column. so lets check if there are no null values in the dataset

In [146]:
data['incident_datetime'].isnull().sum()

0

We can confirm that there are no null values.

We can sort the data set with the incident datetime column

In [147]:
data.sort_values(by='incident_datetime', inplace=True)

Now we can be sure that the dataset has been sorted

**Now we want to create two different column from the datetime column, one of the date (Year-Month-Day) and another for the time zone (Hour-minute-Seconds)**

In [148]:
# Creating a new column of just the date
# Creating a new column of just the time
data['incident_date'] = data['incident_datetime'].str[:10]
data['incident_time'] = data['incident_datetime'].str[10:]


Now we can drop/del the incident_datetime column and also rearrange the data bring the newly created columns to the front

In [149]:

# we use the pop function
first_col = data.pop('incident_date')
second_col = data.pop('incident_time')

# then we insert it the column number position, the name of the columns, and the coloumn we popped
data.insert(1, 'Incident_date', first_col)
data.insert(2, 'Incident_time', second_col)

# we can use the drop function to delete the incident datetime column
data.drop('incident_datetime', axis=1)


acc_code Incident_date Incident_time      report_datetime  \
0        571905    2019-01-01      00:00:00  2019-01-02 06:11:00   
1       3790870    2019-01-01      00:03:00  2020-02-20 13:48:00   
2        599075    2019-01-01      00:05:00  2019-01-01 10:35:00   
3        571924    2019-01-01      00:20:00  2019-01-02 05:12:00   
4        599523    2019-01-01      00:25:00  2019-01-04 09:42:00   
...         ...           ...           ...                  ...   
81730   7534454    2022-12-31      23:11:00  2023-01-01 21:49:00   
81731   7534701    2022-12-31      23:30:00  2023-01-01 22:55:00   
81732   7567022    2022-12-31      23:30:00  2023-01-23 09:17:00   
81733   7531271    2022-12-31      23:45:00  2023-01-01 05:36:00   
81734   7556715    2022-12-31      23:55:00  2023-01-11 14:50:00   

                                         province_th          province_en  \
0                                 ลพบุรี               Loburi   
1                  อุบลราชธานี     Ubon Ratchathani   
2      ประจวบคีรีขันธ์  Prachuap Khiri Khan   
3                        เชียงใหม่           Chiang Mai   
4                        นครสวรรค์         Nakhon Sawan   
...                                              ...                  ...   
81730                          นนทบุรี           Nonthaburi   
81731                                   ตราด                 Trat   
81732                    มหาสารคาม        Maha Sarakham   
81733                    กำแพงเพชร       Kamphaeng Phet   
81734                       เชียงราย           Chiang Rai   

                          agency  \
0      department of rural roads   
1         department of highways   
2         department of highways   
3      department of rural roads   
4         department of highways   
...                          ...   
81730     department of highways   
81731     department of highways   
81732     department of highways   
81733     department of highways   
81734     department of highways   

                                                   route  \
0      แยกทางหลวงหมายเà...   
1      เดชอุดม - อุบลราชà...   
2      ปราณบุรี - ปากน้ำà...   
3      เชื่อมทางหลวงหมà...   
4             เกยไชย - ศรีมงคล   
...                                                  ...   
81730  แคราย - คลองบ้านใà...   
81731               ตราด - แหลมงอบ   
81732  วาปีปทุม - พยัคฆภà...   
81733         นครชุม - มอเจริญ   
81734  แยกแม่กรณ์ - ห้วยà...   

                vehicle_type                                 presumed_cause  \
0                 motorcycle         driving under the influence of alcohol   
1      private/passenger car                                       speeding   
2                 motorcycle                                       speeding   
3                 motorcycle         driving under the influence of alcohol   
4      private/passenger car  cutting in closely by people/vehicles/animals   
...                      ...                                            ...   
81730             motorcycle                                       speeding   
81731             motorcycle                                       speeding   
81732                    van                                       speeding   
81733             motorcycle         driving under the influence of alcohol   
81734  private/passenger car                                       speeding   

                            accident_type  number_of_vehicles_involved  \
0                                   other                            1   
1        rollo

### **Report_datetime column**

In [151]:
data['report_datetime']

0        2019-01-02 06:11:00
1        2020-02-20 13:48:00
2        2019-01-01 10:35:00
3        2019-01-02 05:12:00
4        2019-01-04 09:42:00
                ...         
81730    2023-01-01 21:49:00
81731    2023-01-01 22:55:00
81732    2023-01-23 09:17:00
81733    2023-01-01 05:36:00
81734    2023-01-11 14:50:00
Name: report_datetime, Length: 81735, dtype: object

In [152]:
data['report_datetime'].info()

<class 'pandas.core.series.Series'>
Int64Index: 81735 entries, 0 to 81734
Series name: report_datetime
Non-Null Count  Dtype 
--------------  ----- 
81735 non-null  object
dtypes: object(1)
memory usage: 1.2+ MB


We can note that the column is an object, so we will change it to a datetime

In [153]:
#code to convert the column to a datetime column
data['report_datetime'] = pd.to_datetime(data['report_datetime'])

In [154]:
data['report_datetime'].info()

<class 'pandas.core.series.Series'>
Int64Index: 81735 entries, 0 to 81734
Series name: report_datetime
Non-Null Count  Dtype         
--------------  -----         
81735 non-null  datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 1.2 MB


### **Province_th Column**

In [155]:
# preview the column
data['province_th']

0                                   ลพบุรี
1                    อุบลราชธานี
2        ประจวบคีรีขันธ์
3                          เชียงใหม่
4                          นครสวรรค์
                             ...                      
81730                            นนทบุรี
81731                                     ตราด
81732                      มหาสารคาม
81733                      กำแพงเพชร
81734                         เชียงราย
Name: province_th, Length: 81735, dtype: object

From the description, this column is the names of the provinces the incidents occurred... I decided to leave the column because if can be use as a dictionary to interpret the english version of the provice name

### **Province_en Column**

In [156]:
#preview the column
data['province_en']

0                     Loburi
1           Ubon Ratchathani
2        Prachuap Khiri Khan
3                 Chiang Mai
4               Nakhon Sawan
                ...         
81730             Nonthaburi
81731                   Trat
81732          Maha Sarakham
81733         Kamphaeng Phet
81734             Chiang Rai
Name: province_en, Length: 81735, dtype: object

Lets check the number of province involved

In [157]:
# LIst of the unique province in the dataset
data['province_en'].unique()

array(['Loburi', 'Ubon Ratchathani', 'Prachuap Khiri Khan', 'Chiang Mai',
       'Nakhon Sawan', 'Mae Hong Son', 'Chumphon', 'Sing Buri',
       'Songkhla', 'Lamphun', 'Trat', 'Phuket', 'Saraburi', 'Ratchaburi',
       'Phra Nakhon Si Ayutthaya', 'Nakhon Ratchasima',
       'Nakhon Si Thammarat', 'Chaiyaphum', 'Kalasin', 'Suphan Buri',
       'Phetchaburi', 'Phrae', 'Chai Nat', 'Prachin Buri',
       'Nakhon Pathom', 'Kanchanaburi', 'Phetchabun', 'Ang Thong',
       'Nonthaburi', 'Samut Prakan', 'Bangkok', 'Phayao', 'Phatthalung',
       'Yala', 'Maha Sarakham', 'Surat Thani', 'Amnat Charoen',
       'Nong Khai', 'Nan', 'Phangnga', 'Narathiwat', 'Samut Sakhon',
       'Chanthaburi', 'Samut Songkhram', 'Phitsanulok', 'Pathum Thani',
       'Tak', 'Loei', 'Chiang Rai', 'Chachoengsao', 'Buri Ram',
       'Uthai Thani', 'Krabi', 'Surin', 'Udon Thani', 'Si Sa Ket',
       'Uttaradit', 'Khon Kaen', 'Kamphaeng Phet', 'Yasothon', 'Satun',
       'Nakhon Nayok', 'Chon Buri', 'Rayong', 'buogkan'

Let convert this column to a categorical column

In [158]:
data['province_en'] = data['province_en'].astype('category')

In [159]:
data['province_en'].info()

<class 'pandas.core.series.Series'>
Int64Index: 81735 entries, 0 to 81734
Series name: province_en
Non-Null Count  Dtype   
--------------  -----   
81735 non-null  category
dtypes: category(1)
memory usage: 721.0 KB


In [160]:
data['province_en'].nunique()

78

* **we can deduce that there are 78 province in which accidents occured in thailand**

In [161]:
data['province_en'].describe()

count       81735
unique         78
top       Bangkok
freq         6439
Name: province_en, dtype: object

* **The most frequent accident occurred in Bangkok province**
* **The number of accident that occurred in Bankok between 2019-2022 is _6439_**

All so from there are some province that are laballed unknown. Lets get the number of the unknown province

In [162]:
(data['province_en']== 'unknown').sum()

34

* **About 34 province in which accident occcurred are unknown**

In [163]:
# These are the information of the unknown columns

data.loc[data['province_en']=='unknown']

acc_code Incident_date Incident_time    incident_datetime  \
501      823206    2019-01-02      21:30:00  2019-01-02 21:30:00   
5913     572732    2019-04-13      18:20:00  2019-04-13 18:20:00   
7643     823207    2019-05-07      03:15:00  2019-05-07 03:15:00   
8039     823208    2019-05-15      11:45:00  2019-05-15 11:45:00   
8044     823209    2019-05-15      13:50:00  2019-05-15 13:50:00   
11279    823210    2019-08-01      08:00:00  2019-08-01 08:00:00   
12264    573720    2019-08-22      23:05:00  2019-08-22 23:05:00   
17787    574701    2019-12-27      16:00:00  2019-12-27 16:00:00   
17810    574704    2019-12-27      16:55:00  2019-12-27 16:55:00   
17957    823211    2019-12-28      02:30:00  2019-12-28 02:30:00   
18344    575103    2019-12-29      03:00:00  2019-12-29 03:00:00   
24473   1306220    2020-04-03      13:30:00  2020-04-03 13:30:00   
26887   1616332    2020-05-26      08:30:00  2020-05-26 08:30:00   
28559   3814651    2020-06-29      13:00:00  2020-06-29 13:00:00   
31644   3814640    2020-08-22      14:00:00  2020-08-22 14:00:00   
32981   3815374    2020-09-13      19:30:00  2020-09-13 19:30:00   
34005   3809425    2020-10-02      00:00:00  2020-10-02 00:00:00   
36919   2602971    2020-11-21      18:27:00  2020-11-21 18:27:00   
40088   2904953    2020-12-31      13:30:00  2020-12-31 13:30:00   
47969   3874490    2021-04-16      08:15:00  2021-04-16 08:15:00   
51157   4557000    2021-06-20      16:30:00  2021-06-20 16:30:00   
53651   4762614    2021-08-21      11:30:00  2021-08-21 11:30:00   
55473   5094302    2021-10-01      07:30:00  2021-10-01 07:30:00   
56059   5094281    2021-10-14      08:00:00  2021-10-14 08:00:00   
56306   5094306    2021-10-18      11:00:00  2021-10-18 11:00:00   
58098   5552224    2021-11-24      07:00:00  2021-11-24 07:00:00   
60086   5461864    2021-12-29      21:30:00  2021-12-29 21:30:00   
60127   5626483    2021-12-30      02:00:00  2021-12-30 02:00:00   
66552   6073483    2022-04-11      04:15:00  2022-04-11 04:15:00   
70656   6584931    2022-06-08      22:00:00  2022-06-08 22:00:00   
73604   7441209    2022-08-04      09:00:00  2022-08-04 09:00:00   
73900   7441295    2022-08-10      00:05:00  2022-08-10 00:05:00   
79539   7442332    2022-11-27      13:30:00  2022-11-27 13:30:00   
79646   7446619    2022-11-29      19:14:00  2022-11-29 19:14:00   

          report_datetime            province_th province_en  \
501   2019-01-03 06:20:00  ไม่ทราบ     unknown   
5913  2019-04-14 11:44:00  ไม่ทราบ     unknown   
7643  2019-05-14 15:37:00  ไม่ทราบ     unknown   
8039  2019-06-10 10:07:00  ไม่ทราบ     unknown   
8044  2019-06-10 10:04:00  ไม่ทราบ     unknown   
11279 2019-09-25 11:26:00  ไม่ทราบ     unknown   
12264 2019-09-02 11:32:00  ไม่ทราบ     unknown   
17787 2019-12-28 06:14:00  ไม่ทราบ     unknown   
17810 2019-12-28 05:48:00  ไม่ทราบ     unknown   
17957 2019-12-29 06:04:00  ไม่ทราบ     unknown   
18344 2020-01-06 10:34:00  ไม่ทราบ     unknown   
24473 2020-05-05 12:50:00  ไม่ทราบ     unknown   
26887 2020-05-29 09:41:00  ไม่ทราบ     unknown   
28559 2021-01-20 07:38:00  ไม่ทราบ     unknown   
31644 2021-01-20 10:22:00  ไม่ทราบ     unknown   
32981 2021-01-25 11:57:00  ไม่ทราบ     unknown   
34005 2020-12-04 14:34:00  ไม่ทราบ     unknown   
36919 2020-11-22 11:31:00  ไม่ทราบ     unknown   
40088 2020-12-31 15:14:00  ไม่ทราบ     unknown   
47969 2021-04-17 05:55:00  ไม่ทราบ     unknown   
51157 2021-08-03 11:29:00  ไม่ทราบ     unknown   
53651 2021-09-05 10:15:00  ไม่ทราบ     unknown   
55473 2021-11-01 10:39:00  ไม่ทราบ     unknown   
56059 2021-11-01 10:32:00  ไม่ทราบ     unknown   
56306 2021-11-01 10:44:00 

below is the recent incident information that happened at the unknown provience